### DISCUSSION TITLE
The car is waking up after a while

### DISCUSSION THREAD

**Discussion by sultantariq** [teslamate-org/teslamate#2891] [Q&A] [answered]
Hi all,

Recently I have noticed that the car is waking up after sometime. The interval is not the same every time.

Is there anyone else who's having the same issue?
The Tesla version is 2022.20.8 while Teslamate version is 1.27.1

Here is a screenshot:
<img width="1234" alt="Screen Shot 2022-09-10 at 13 16 47" src="https://user-images.githubusercontent.com/42979273/189480814-923f0b9c-09d0-40c3-a940-348914cfffd8.png">

Logs:

> teslamate_1  | 2022-09-09 21:13:06.588 car_id=1 [info] Start / :asleep
teslamate_1  | 2022-09-09 21:13:06.594 car_id=1 [info] Disconnecting ...
teslamate_1  | 2022-09-09 21:31:20.537 [info] Refreshing access token ...
teslamate_1  | 2022-09-09 21:31:21.027 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (488.845 ms)
teslamate_1  | 2022-09-09 21:31:21.038 [info] Scheduling token refresh in 6 h
teslamate_1  | 2022-09-09 22:00:27.532 car_id=1 [info] Start / :online
teslamate_1  | 2022-09-09 22:00:27.561 car_id=1 [info] Connecting ...
teslamate_1  | 2022-09-09 22:03:30.571 car_id=1 [info] Suspending logging
teslamate_1  | 2022-09-09 22:13:38.050 car_id=1 [info] Fetching vehicle state ...
teslamate_1  | 2022-09-09 22:13:38.340 car_id=1 [info] Start / :asleep
teslamate_1  | 2022-09-09 22:13:38.349 car_id=1 [info] Disconnecting ...
teslamate_1  | 2022-09-10 03:31:21.039 [info] Refreshing access token ...
teslamate_1  | 2022-09-10 03:31:21.332 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (291.851 ms)
teslamate_1  | 2022-09-10 03:31:21.342 [info] Scheduling token refresh in 6 h
teslamate_1  | 2022-09-10 08:01:55.466 car_id=1 [info] Start / :online
teslamate_1  | 2022-09-10 08:01:55.497 car_id=1 [info] Connecting ...
teslamate_1  | 2022-09-10 08:05:00.348 car_id=1 [info] Suspending logging
teslamate_1  | 2022-09-10 08:14:08.513 [error] Vehicle Error: Vehicle is offline
teslamate_1  | 2022-09-10 08:15:25.817 car_id=1 [info] Fetching vehicle state ...
teslamate_1  | 2022-09-10 08:15:26.112 car_id=1 [info] Start / :asleep
teslamate_1  | 2022-09-10 08:15:26.119 car_id=1 [info] Disconnecting ...
teslamate_1  | 2022-09-10 08:49:15.440 car_id=1 [info] Start / :online
teslamate_1  | 2022-09-10 08:49:15.473 car_id=1 [info] Connecting ...
teslamate_1  | 2022-09-10 08:51:25.898 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/**************/vehicle_data -> 408 (9080.734 ms)
teslamate_1  | 2022-09-10 08:51:25.898 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
teslamate_1  | 2022-09-10 08:51:25.899 car_id=1 [error] Error / :unknown
teslamate_1  | 2022-09-10 08:51:55.009 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/**************/vehicle_data -> 408 (9108.148 ms)
teslamate_1  | 2022-09-10 08:51:55.010 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}
teslamate_1  | 2022-09-10 08:51:55.010 car_id=1 [error] Error / :unknown
teslamate_1  | 2022-09-10 08:52:23.285 car_id=1 [info] Suspending logging
teslamate_1  | 2022-09-10 09:02:29.959 car_id=1 [info] Fetching vehicle state ...
teslamate_1  | 2022-09-10 09:02:30.241 car_id=1 [info] Start / :asleep
teslamate_1  | 2022-09-10 09:02:30.247 car_id=1 [info] Disconnecting ...
teslamate_1  | 2022-09-10 09:31:21.343 [info] Refreshing access token ...
teslamate_1  | 2022-09-10 09:31:21.677 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (333.072 ms)
teslamate_1  | 2022-09-10 09:31:21.686 [info] Scheduling token refresh in 6 h
teslamate_1  | 2022-09-10 09:55:54.674 car_id=1 [info] Start / :online
teslamate_1  | 2022-09-10 09:55:54.704 car_id=1 [info] Connecting ...
teslamate_1  | 2022-09-10 09:58:57.910 car_id=1 [info] Suspending logging
teslamate_1  | 2022-09-10 10:08:06.461 [error] Vehicle Error: Vehicle is offline
teslamate_1  | 2022-09-10 10:09:23.668 car_id=1 [info] Fetching vehicle state ...
teslamate_1  | 2022-09-10 10:09:24.835 car_id=1 [info] Start / :asleep
teslamate_1  | 2022-09-10 10:09:24.846 car_id=1 [info] Disconnecting ...
teslamate_1  | 2022-09-10 10:40:09.427 [info] GET /
teslamate_1  | 2022-09-10 10:40:09.445 [info] Sent 200 in 17ms
teslamate_1  | 2022-09-10 10:57:47.830 car_id=1 [info] Start / :online
teslamate_1  | 2022-09-10 10:57:47.864 car_id=1 [info] Connecting ...
teslamate_1  | 2022-09-10 11:00:53.802 car_id=1 [info] Suspending logging
teslamate_1  | 2022-09-10 11:00:54.734 car_id=1 [warning] Received stale stream data: %TeslaApi.Stream.Data{elevation: 28, est_heading: 4, est_lat: *******, est_lng: *****, est_range: 166, heading: 6, odometer: 4852.7, power: 0, range: 190, shift_state: nil, soc: 62, speed: nil, time: ~U[2022-09-10 11:00:52.954Z]}

**Comment by SorenLan** [comment]
Yes, that is for me also. Started a couple of weeks ago. Periodically wakeups in daytime, sleeping at nights. I changed my Tesla account password and only logged in TeslaMate and the Tesla mobile app after that so that I could be sure that it was not any other app that was waking the car. The problem still occured. The car has version 2022.24.5, Teslamate 1.27.1 and mobile app for iPhone 4.12.1

**Reply by Pengphil** [reply]
same as your case，the problem seems like occured after updating the Tesla version 2022.20.8，I changed my Tesla password ，restarted the docker containers ，but still not resolved 

![image](https://user-images.githubusercontent.com/49644232/190043029-905f8f2c-3f33-4d76-b8b7-c97fcadb4e7c.png)

**Reply by SorenLan** [reply]
Checked my update history. Got 2022.20.8 August 10th. Still sleeping daytime after that. Got 2022.24.5 August 24th and after that I got constant daytime wakeups.

**Reply by sultantariq** [reply]
I have got the new update for Tesla that is [2022.24.6](https://www.notateslaapp.com/software-updates/version/2022.24.6/release-notes) but I am having the same issue.

I have dig a bit deep and found out that Tesla iOS (version 4.12.1) is working in background in those time when Tesla is getting up from sleep. Can you guys confirm this?

![IMG_3785](https://user-images.githubusercontent.com/42979273/190105724-11fc9c11-9228-490d-b7c6-502c9dbf9041.jpg)
![IMG_3786](https://user-images.githubusercontent.com/42979273/190105769-f4700885-39ff-4b86-b154-5415e08b2220.PNG)

**Reply by Pengphil** [reply]
It is a good point，I will kill the Tesla app  (iOS version 4.12.1) background today, and then take a look

**Comment by Pengphil** [comment]
Turning off the Tesla app background refresh settings seems to work
![1663227278043](https://user-images.githubusercontent.com/49644232/190343503-f4bd168c-f8e7-4ada-88c3-fdf98ff703fc.png)

**Reply by achint-s** [reply]
This worked for me for a day but then back to the original issue - car is waking up every hour (or at night once every 2 hours). I have disabled background refresh and also killed the app. Didnt help. Only affects car no. 1 - no issues with car no. 2.

**Reply by Pengphil** [reply]
Has anyone else installed the Tesla app? I disabled background refresh ,it didn't wake up last night

**Comment by lynndixon** [comment]
Same here. Car has been waking and sleeping every 40 minutes or so. It will wake for 20 minutes, then sleep for about 45 minutes then wake, and repeat.

Updated Tesla iOS app, and have changed my password on Tesla.com to invalidate tokens. I am using Teslafi and Teslamate, so I am going to test by ONLY authing Teslafi for a bit, then switching to Teslamate to see if one is causing the issue.

**Reply by Pengphil** [reply]
Has anyone else installed the Tesla app? I disabled background refresh ,it didn't wake up last night

**Reply by SorenLan** [reply]
No more wakeups after I disabled Background refresh.

**Reply by lynndixon** [reply]
So,  I reset my Tesla password, which invaldiated my tokens.  The car slept perfectly fine for 4-5 hours like normal.  Then I decided to setup a token for Teslafi, and the car has been sleeping for 14 hours with zero issues.

Next, I am going to use that same token for Teslamate and see how it goes.

**Comment by Str1atum** [comment]
Stopped with 2022.24 for me

**Comment by sultantariq** [comment, marked as answer]
So the issue is fixed when I got the new updates for iOS and Tesla. 
iOS version : 4.13.1
Tesla version : 2022.28.1

**Reply by mikephung** [reply]
hi @sultantariq, thank you for the reply, do you have a facelift or pre-facelift version?
